src/Maestro/tests/DependencyUpdateErrorProcessor.Tests/DepenedencyUpdateErrorProcessorTests.cs

```diff
@@ -0,0 +1,94 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using Maestro.Data;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.Dotnet.GitHub.Authentication;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.ServiceFabric.Data;
+using Moq;
+using Octokit;
+using ServiceFabricMocks;
+
+namespace DependencyUpdateErrorProcessor.Tests
+{
+    public class DependencyUpdateErrorProcessorTests : IDisposable
+    {
+        private readonly Lazy<TestBuildAssetRegistryContext> _context;
+        protected readonly Mock<IHostingEnvironment> Env;
+        protected readonly ServiceProvider Provider;
+        protected readonly IServiceScope Scope;
+        protected readonly MockReliableStateManager StateManager;
+        protected readonly Mock<GitHubClient> GithubClient;
+
+        public DependencyUpdateErrorProcessorTests()
+        {
+            var services = new ServiceCollection();
+            StateManager = new MockReliableStateManager();
+            Env = new Mock<IHostingEnvironment>(MockBehavior.Strict);
+            GithubClient = new Mock<GitHubClient>(null);
+            services.AddSingleton(Env.Object);
+            services.AddSingleton<IReliableStateManager>(StateManager);
+            services.AddLogging();
+            services.AddDbContext<TestBuildAssetRegistryContext>(
+                options => { options.UseInMemoryDatabase("BuildAssetRegistry"); });
+            services.AddSingleton<Func<string,GitHubClient>>(
+             repo =>
+             {
+                 return GithubClient.Object;
+             });
+            services.Configure<DependencyUpdateErrorProcessorOptions>(
+                (options) =>
+                {
+                    options.IsEnabled = true;
+                    options.FyiHandle = "@epananth";
+                    options.GithubUrl = "https://github.com/maestro-auth-test/maestro-test2";
+                });
+            Provider = services.BuildServiceProvider();
+            Scope = Provider.CreateScope();
+            _context = new Lazy<TestBuildAssetRegistryContext>(GetContext);
+            SetUp();
+        }
+
+        public TestBuildAssetRegistryContext Context => _context.Value;
+
+        private TestBuildAssetRegistryContext GetContext()
+        {
+            return Scope.ServiceProvider.GetRequiredService<TestBuildAssetRegistryContext>();
+        }
+
+        public void Dispose()
+        {
+            Env.VerifyAll();
+            Scope.Dispose();
+            Provider.Dispose();
+        }
+
+        public void SetUp()
+        {
+            var repositoryBranchUpdate = new RepositoryBranchUpdateHistoryEntry()
```

Comment:
Why is there a bunch of code in the constructor AND a bunch of code in SetUp? It seems like only one of the two should be necessary.

Reply:
Setup code will go to a Fact in my next PR. I didn't know where to put this code. So I added it as part of setup. Is that ok?

Reply:
It's just odd the constructor is big, and this method is big.  Why not just have all the code in "SetUp"?

Reply:
Since I will be moving the code inside my fact( in my next PR). I will just remove setup altogether, we don't actually need that code for now. I will put the setup in a different file in my next PR. That way we only have one constructor.